src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -397,7 +397,7 @@ void mf::WlSurface::commit(WlSurfaceState const& state)
             stream->submit_buffer(mir_buffer);
             auto const new_buffer_size = stream->stream_size();
 
-            if (!input_shape && std::make_optional(new_buffer_size) != buffer_size_)
```

Comment:
This looks wrong: if the client has set an input shape, then why would we recalculate it?

Reply:
In my case, I was getting an `input_shape` first and _then_ a `new_buffer_size` afterward. Because pointer events are dependent on the intersection between these two, it made sense to me that we update whenever the `new_buffer_size` comes through